presto-native-execution/presto_cpp/main/PeriodicServiceInventoryManager.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/io/async/EventBaseThread.h>
+#include <presto_cpp/main/http/HttpClient.h>
+#include "presto_cpp/main/CoordinatorDiscoverer.h"
+
+namespace facebook::presto {
+
+class PeriodicServiceInventoryManager {
```

Comment:
Can we name it AnnoucementManager? Its not maintaining service, just publishing some stats.

Reply:
service inventory tier name comes from presto java. https://fburl.com/code/y8vf5vup